diff --git a/tests/conftest.py b/tests/conftest.py
index a13e1de..431f2fd 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -56,28 +56,28 @@ def get_mock_job():
             job_parameters = {'param1': 'value1'}
         if job_id is None:
             job_id = str(uuid4())
-        job_dict = {
-            'job_type': job_type,
-            'job_id': job_id,
-            'request_time': request_time.isoformat(timespec='seconds'),
-            'status_code': status_code,
-            'user_id': user_id,
-            'name': name,
-            'job_parameters': job_parameters,
-            'files': files,
-            'browse_images': browse_images,
-            'thumbnail_images': thumbnail_images,
-            'expiration_time': expiration_time,
-        }
-        keys_to_delete = []
-        for k, v in job_dict.items():
-            if v is None:
-                keys_to_delete.append(k)
-
-        for key in keys_to_delete:
-            del job_dict[key]
-
-        return Job.from_dict(job_dict)
+
+        job = Job(
+            job_type=job_type,
+            job_id=job_id,
+            request_time=request_time,
+            status_code=status_code,
+            user_id=user_id,
+            name=name,
+            bucket=bucket,
+            bucket_prefix=bucket_prefix,
+            job_parameters=job_parameters,
+            files=files,
+            logs=logs,
+            browse_images=browse_images,
+            thumbnail_images=thumbnail_images,
+            expiration_time=expiration_time,
+            processing_times=processing_times,
+            credit_cost=credit_cost,
+            priority=priority,
+        )
+
+        return job
 
     return default_job
 
